test/e2e/fixture/applicationsets/utils/fixture.go

```diff
@@ -113,33 +113,48 @@ func EnsureCleanState(t *testing.T) {
 
 	policy := v1.DeletePropagationForeground
 
-	// Delete the applicationset-e2e namespace, if it exists
-	err := fixtureClient.KubeClientset.CoreV1().Namespaces().Delete(context.Background(), ApplicationsResourcesNamespace, v1.DeleteOptions{PropagationPolicy: &policy})
-	if err != nil && !strings.Contains(err.Error(), "not found") { // 'not found' error is expected
-		CheckError(err)
-	}
-
-	// Delete the argocd-e2e-external namespace, if it exists
-	err2 := fixtureClient.KubeClientset.CoreV1().Namespaces().Delete(context.Background(), string(ArgoCDExternalNamespace), v1.DeleteOptions{PropagationPolicy: &policy})
-	if err2 != nil && !strings.Contains(err2.Error(), "not found") { // 'not found' error is expected
-		CheckError(err2)
-	}
-
-	// Delete the argocd-e2e-external namespace, if it exists
-	err3 := fixtureClient.KubeClientset.CoreV1().Namespaces().Delete(context.Background(), string(ArgoCDExternalNamespace2), v1.DeleteOptions{PropagationPolicy: &policy})
-	if err3 != nil && !strings.Contains(err3.Error(), "not found") { // 'not found' error is expected
-		CheckError(err3)
-	}
-
-	// delete resources
-	// kubectl delete applicationsets --all
-	CheckError(fixtureClient.AppSetClientset.DeleteCollection(context.Background(), v1.DeleteOptions{PropagationPolicy: &policy}, v1.ListOptions{}))
-	// kubectl delete apps --all
-	CheckError(fixtureClient.AppClientset.ArgoprojV1alpha1().Applications(TestNamespace()).DeleteCollection(context.Background(), v1.DeleteOptions{PropagationPolicy: &policy}, v1.ListOptions{}))
-
-	// kubectl delete secrets -l e2e.argoproj.io=true
-	CheckError(fixtureClient.KubeClientset.CoreV1().Secrets(TestNamespace()).DeleteCollection(context.Background(),
-		v1.DeleteOptions{PropagationPolicy: &policy}, v1.ListOptions{LabelSelector: TestingLabel + "=true"}))
+	fixture.RunFunctionsInParallelAndCheckErrors(t, map[string]func() error{
+		"delete_applications_resources_namespace": func() error {
+			// Delete the applicationset-e2e namespace, if it exists
+			err := fixtureClient.KubeClientset.CoreV1().Namespaces().Delete(context.Background(), ApplicationsResourcesNamespace, v1.DeleteOptions{PropagationPolicy: &policy})
+			if err != nil && !strings.Contains(err.Error(), "not found") { // 'not found' error is expected
```

Comment:
I believe there's a k8s library function that checks whether an error is a "not found" error.